### PR TITLE
DBZ-1279 Dockerfile improvements

### PR DESCRIPTION
**Pull request by renatomefi**
# Context

General improvements among CI, Dockerfiles and Shell scripts

## Changes

Review per commit is advised!

- Seems like there's a bug in the Jenkins job where it's copying the `0.9` dir in the `0.10` one, I've removed the extra files
- Implement Hadolint: https://github.com/hadolint/hadolint, the individual checks are  documented here: https://github.com/hadolint/hadolint/wiki
- Fix a series of the things proposed by the linter
- Improve build scripts
- shellcheck build scripts
- shellcheck container scripts - Not anymore, it's looking too risky with a series of things that can go wrong

### CODE REVIEW

**Comment by jpechane** [comment]
@renatomefi Thanks for the PR, I'll take a look in detail later. Just one proposal I think it makes sense to update only 0.9 and 0.10 Dockerfiles. I believe it is unnecessary to update the old versions.

**Comment by renatomefi** [comment]
@jpechane indeed, can I keep the changes I've already done or you'd like me to revert them?
Also, I was planning to fix the entrypoint scripts, but it's too risky, there are many, many things going on there and since there are no tests I think I'll leave it as it is.
So this PR is ready to review now!

**Comment by renatomefi** [comment]
@jpechane also, I don't have access to the Jenkinsfile, but certainly there's a bug there which is copying the new releases wrongly, would you check that please? thanks

**Comment by jpechane** [comment]
@renatomefi FYI - https://github.com/debezium/debezium/blob/master/jenkins-jobs/scripts/release-pipeline.groovy

**Comment by gunnarmorling** [comment]
Hey @renatomefi; I think we should only change the 0.9 and 0.10 images. The older ones are not maintained any more and I'd prefer to avoid the perception of that by changing their images.

File: connect/0.10/docker-debezium-download.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
```
Comment (jpechane) [inline connect/0.10/docker-debezium-download.sh]: @renatomefi Does it make sense to make a single library shell script file located in `connect-base` that would contain functions for obtaining deps from Maven central and confluent and alss function to download debezium connectors? Then other scripts will just source this library file un use the functions.

Comment (renatomefi) [reply connect/0.10/docker-debezium-download.sh]: @jpechane I think it makes sense.
I can also enforce the environment variables to be there

**Comment by gunnarmorling** [comment]
Already merged the first commit ("DBZ-1279 Remove incidental 0.9 version within 0.10"), as this one was indisputable in any case. Thanks a lot, @renatomefi!

**Comment by gunnarmorling** [comment]
@renatomefi Could you (force-) push to this PR so that the already merged commit gets removed from it, showing just the diff with out that one? Thanks!

**Comment by gunnarmorling** [comment]
@renatomefi ping :) Could you do the push as suggested above? Also the "WIP" in the last commit message seems to suggest you're planning to add more things here?

File: check-all.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+##
+## Lints all Dockerfiles with Hadolint
+##
+docker run --rm -i -v "$(pwd)":/p --workdir=/p \
+  hadolint/hadolint:latest-debian \
```
Comment (gunnarmorling) [inline check-all.sh]: Nice, didn't know such thing exists!

File: connect-base/0.10/Dockerfile [hunk 3]
```diff
@@ -28,20 +28,21 @@ RUN mkdir $KAFKA_CONNECT_PLUGINS_DIR;
 # This will prevent the classes in the shared dependencies from appearing in multiple JARs
 # on the classpath, which results in arcane NoSuchMethodError exceptions.
 #
-
-RUN cd $KAFKA_HOME/libs &&\
-    function confluent_dep { curl -O http://packages.confluent.io/maven/io/confluent/$1/$CONFLUENT_VERSION/$1-$CONFLUENT_VERSION.jar;  echo "$2 $1-$CONFLUENT_VERSION.jar" >> sums.md5; };\
-    function central_dep { curl -O http://central.maven.org/maven2/$1/$2/$3/$2-$3.jar; echo "$4 $2-$3.jar" >> sums.md5; };\
+# hadolint ignore=DL3003
```
Comment (gunnarmorling) [inline connect-base/0.10/Dockerfile]: Could you add a comment what DL3003 is about?

Comment (gunnarmorling) [reply connect-base/0.10/Dockerfile]: Pushed a commit for that.

**Comment by gunnarmorling** [comment]
Superseded by #130.

**Pull request closed without merging**